### PR TITLE
Prevent NPE on notification email

### PR DESCRIPTION
**Pull request by HemKal**
Check for the form objects which don't carry mail in the payload

#968 NPE on Submitting PCA form.

### CODE REVIEW

**Comment by jasonaowen** [comment]
Jenkins, test this please.

**Comment by jasonaowen** [comment]
@HemKal, please fix [the Checkstyle warnings](https://ci.psm.solutionguidance.com/job/checkstyle/988/checkstyleResult/new/); you can run Checkstyle locally with `./gradlew checkstyleMain checkstyleTest`.

**Comment by jasonaowen** [comment]
Jenkins, test this please.

**Comment by HemKal** [comment]
Jenkins, test this please.

**Comment by jasonaowen** [comment]
Jenkins, test this please.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java [hunk 1]
```diff
@@ -72,10 +71,7 @@ private VelocityEngine getVelocityEngine() {
         return velocityEngine;
     }
 
-    private String populateTemplate(
-        EmailTemplate emailType,
-        Map<String, Object> vars
-    ) {
+    private String populateTemplate(EmailTemplate emailType, Map<String, Object> vars) {
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java]: It looks like you did an automatic reformat of this file, which led to several changes that aren't relevant to your intended change. In particular, it also reformatted this and other method signatures into long lines, which are not yet enforced by checkstyle.

Please remove these irrelevant changes from your commit.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java [hunk 2]
```diff
@@ -87,11 +83,7 @@ private String populateTemplate(
         return writer.toString();
     }
 
-    private void saveNotification(
-        String emailAddress,
-        EmailTemplate emailType,
-        Map<String, Object> vars
-    ) {
+    private void saveNotification(String emailAddress, EmailTemplate emailType, Map<String, Object> vars) {
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java]: As above, this change is not relevant, please remove it.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java [hunk 3]
```diff
@@ -119,33 +111,32 @@ private void saveNotification(
...
-      MimeMessage message = new MimeMessage(session);
-      try {
-          message.setRecipient(RecipientType.TO, new InternetAddress(emailAddress));
-          message.setSubject(config.getEmailSubject(emailType));
-          message.setText(populateTemplate(emailType, vars));
-
-          saveNotification(emailAddress, emailType, vars);
-          Transport.send(message);
+    public void sendNotification(String emailAddress, EmailTemplate emailType, Map<String, Object> vars)
+            throws PortalServiceException {
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java]: Please don't reformat the method signature; it should only change if you need to change the return type, the arguments, or the `throws` clause, and then only the relevant line(s) should change.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java [hunk 4]
```diff
@@ -119,33 +111,32 @@ private void saveNotification(
...
+        try {
+            message.setRecipient(RecipientType.TO, new InternetAddress(emailAddress));
+            message.setSubject(config.getEmailSubject(emailType));
+            message.setText(populateTemplate(emailType, vars));
+
+            saveNotification(emailAddress, emailType, vars);
+            Transport.send(message);
         } catch (Exception e) {
-          throw new PortalServiceException("Error while sending notification.", e);
+            throw new PortalServiceException("Error while sending notification.", e);
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java]: While this whitespace change does bring this line into compliance with our 4-space indenting policy, it's not relevant to the change you're trying to make; please either remove the change or split it into its own whitespace-only commit.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java [hunk 5]
```diff
@@ -119,33 +111,32 @@ private void saveNotification(
...
         }
     }
 
     @Override
-    public void sendEnrollmentNotification(
-        EnrollmentType enrollment,
-        EmailTemplate emailType
-    ) throws PortalServiceException {
+    public void sendEnrollmentNotification(EnrollmentType enrollment, EmailTemplate emailType)
+            throws PortalServiceException {
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/NotificationServiceBean.java]: Please don't reformat the method signature.

**Comment by jasonaowen** [comment]
@HemKal if you don't have bandwidth or are not comfortable with the kind of commit-editing I'm asking for, I'd be happy to help! That can either be pairing or just having me do it, whichever you prefer.

**Pull request closed without merging**